forge-gui/res/cardsfolder/upcoming/hag_of_ceaseless_torment.txt

```diff
@@ -0,0 +1,12 @@
+Name:Hag of Ceaseless Torment
+ManaCost:2 B
+Types:Creature Hag Warlock
+PT:2/2
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigRepeat | TriggerDescription$ At the beginning of your upkeep, each opponent loses 3 life unless that player sacrifices a nonland permanent or discards a card.
+SVar:TrigRepeat:DB$ RepeatEach | RepeatPlayers$ Opponent | RepeatSubAbility$ DBChoose
+SVar:DBChoose:DB$ GenericChoice | Defined$ Player.IsRemembered | Choices$ SacNonland,Discard | FallbackAbility$ LoseLifeFallback | AILogic$ PayUnlessCost
```

Comment:
I think you can use `Defined$ Remembered` here and line 10 since the only thing remembered at this point is the player in question

Reply:
Would this apply to line 8 and 9 too?